### PR TITLE
Revert Windows GPU python 3.8 and python 3.9 changes in the release branch

### PR DESCRIPTION
**Pull request by snnn**
**Description**: 

Revert python 3.8 changes
1.  Revert "Revert some pipeline changes in #5874"
2.  Revert "Add python 3.9 support (#5874)"
3.  Revert "Add support for Python 3.8+ on Windows when CUDA is enabled (#5956)"
4.  Cherry-pick #5990 to validate the changes.
5.  Regenerate all the CI build docker images to ensure things are consistent. 
**Motivation and Context**
- Why is this change required? What problem does it solve?
- If it fixes an open issue, please link to the issue here.

Will need to reopen #5697 #4035 #2664 #5963

### CODE REVIEW

File: tools/ci_build/github/azure-pipelines/templates/py-packaging-stage.yml [hunk 1]
```diff
@@ -158,8 +156,6 @@ stages:
             python.version: '3.7'
           Python38:
             python.version: '3.8'
```
Comment (duli2012) [inline tools/ci_build/github/azure-pipelines/templates/py-packaging-stage.yml:158]: python 3.8 is still here?

File: tools/ci_build/github/azure-pipelines/templates/py-packaging-stage.yml [hunk 2]
```diff
@@ -238,8 +234,6 @@ stages:
...
```
Comment (duli2012) [inline tools/ci_build/github/azure-pipelines/templates/py-packaging-stage.yml:236]: same as above

**Comment by jywu-msft** [comment]
what do you think of just respecting CUDA_PATH as I suggested in https://github.com/microsoft/onnxruntime/issues/5962#issuecomment-738944475
rather than hardcoding CUDA 10.2?

File: tools/ci_build/github/azure-pipelines/templates/py-packaging-stage.yml (hunk 2 above)
Comment (snnn) [reply tools/ci_build/github/azure-pipelines/templates/py-packaging-stage.yml:236]: It was added long ago. In the past a few releases, we have almost full support for python 3.8 except lacking Windows GPU package and ARM package.

**Comment by snnn** [comment]
> what do you think of just respecting CUDA_PATH as I suggested in [#5962 (comment)](https://github.com/microsoft/onnxruntime/issues/5962#issuecomment-738944475)
> rather than hardcoding CUDA 10.2?

Sorry I have no idea.  I think the safest way would be revert the recent changes first, get the release branch back to a good state, then we can think how to do it better.

**Comment by jywu-msft** [comment]
> > what do you think of just respecting CUDA_PATH as I suggested in [#5962 (comment)](https://github.com/microsoft/onnxruntime/issues/5962#issuecomment-738944475)
> > rather than hardcoding CUDA 10.2?
> 
> Sorry I have no idea. I think the safest way would be revert the recent changes first, get the release branch back to a good state, then we can think how to do it better.

I have a fix soon (hopefully). Let me share a branch with you. Let me know what you think.

**Review by duli2012 — APPROVED** [review]

**Review by yufenglee — APPROVED** [review]

**Pull request merged**